File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestionsDomain/src/main/java/org/octo/seatingplacessuggestions/domain/seatingplacesuggestions/SuggestionIsMade.java
package org.octo.seatingplacessuggestions.domain.seatingplacesuggestions;

import java.util.List;

public class SuggestionIsMade {

    private final List<SeatingPlace> suggestedSeats;
    private final PartyRequested partyRequested;
    private final PricingCategory pricingCategory;

    public SuggestionIsMade(SeatingOptionIsSuggested seatingOptionIsSuggested) {
        this.suggestedSeats = seatingOptionIsSuggested.seats();
        this.partyRequested = seatingOptionIsSuggested.partyRequested();
        this.pricingCategory = seatingOptionIsSuggested.pricingCategory();
    }

    public List<String> seatNames() {
        return suggestedSeats.stream().map(SeatingPlace::toString).toList();
    }

    public boolean matchExpectation() {
        return suggestedSeats.size() == partyRequested.partySize();
    }

    public PricingCategory pricingCategory() {
        return pricingCategory;
    }
}

File: TheaterSuggestions/CSharp/TheaterSuggestions.Tests/SuggestionIsMade.cs
using System.Collections.Generic;
using System.Linq;

namespace SeatsSuggestions.Tests;

public class SuggestionIsMade(SeatingOptionIsSuggested seatingOptionIsSuggested)
{
    private readonly List<SeatingPlace> _suggestedSeats = seatingOptionIsSuggested.Seats;

    public int PartyRequested { get; } = seatingOptionIsSuggested.PartyRequested;
    public PricingCategory PricingCategory { get; } = seatingOptionIsSuggested.PricingCategory;

    public IReadOnlyList<SeatingPlace> SuggestedSeats => _suggestedSeats;

    public IEnumerable<string> SeatNames()
    {
        return _suggestedSeats.Select(s => s.ToString());
    }

    public bool MatchExpectation()
    {
        return _suggestedSeats.Count == PartyRequested;
    }
}